### ISSUE TITLE
binary data in new version

### ISSUE THREAD

**Issue by darkjacky** [SuperiorServers/gm_tmysql4#19]
The new version doesn't handle data that isn't a strings well.
Put some data in a database using https://github.com/darkjacky/gm_binarytable for example and it will not come back as the same data.
The previous version handled it fine.

**Comment by KingofBeast** [comment]
@darkjacky does the data store correctly and only retrieve incorrectly? Which side of that does the issue lie on?

**Comment by darkjacky** [comment]
I did not try storing data in the new version. The data was already on the SQL table and it gets fetched incorrectly.
Perhaps it will also store it wrong but again I did not try as things broke as soon as I installed it.

**Comment by KingofBeast** [comment]
I don't believe any code in the module itself changed in a way to make this regression possible so I'm suspicious of the MariaDB connector. I will toy with different versions of the connectors and report back. Do you have any info on whether the fault is solely on Windows or Linux, or both?

**Comment by darkjacky** [comment]
This is caused by my table being a default of latin1 rather than utf8mb4. For some reason the latest version can't translate that while earlier versions could. It has been fixed now by changing the database to utf8mb4. Though it would be better if the module could handle that.
